### PR TITLE
add-ASimFileEventMicrosoftAuditEvent

### PR DESCRIPTION
**Pull request by KustoKing**
Required items, please complete
   
   Change(s):
   - Add ASimFileEventMicrosoftAuditEvent

   Reason for Change(s):
   - New ASim parser

   Version Updated:
   - Yes Version and date

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

File: Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    , ActorUserIdType = iif(SubjectUserSid startswith_cs "S-1-5-","SID","")
+    , ActorUsernameType = iif(Account contains "\\", "Windows", "")
+    , ActorUserType = iif(AccountType == "Machine", "Machine", "Regular")
+    , EventStartTime = TimeGenerated
+    , EventEndTime = TimeGenerated
+    , TargetFilePath = iif(ObjectName startswith @"\Device\",""
+      ,ObjectName) 
+    , TargetFilePathFormat = iif(ObjectName !startswith @"\Device\","Windows Local"
+      ,"")
+    , FilePath = iif(ObjectName startswith @"\Device\",""
```
Comment (oshezaf) [inline Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml:56]: No point in calulating again. Just assign from TargetFilePath.

File: Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    , AccessMask == "0x4", "ObjectModified"
+    , AccessMask == "0x40", "ObjectDeleted"
+    , AccessMask == "0x40000", "MetadataModified"
+    , AccessMask == "0x6", "ObjectModified"
+    , AccessMask == "0x8", "MetadataAccessed"
+    , AccessMask == "0x80", "MetadataAccessed"
+    , AccessMask == "0x80000", "MetadataModified"
+    , "")
+    , ActingProcessName = ProcessName
+    , ActorUserIdType = iif(SubjectUserSid startswith_cs "S-1-5-","SID","")
```
Comment (oshezaf) [inline Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml:47]: I assume it is always SID.

File: Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    , TargetFilePathFormat = iif(ObjectName !startswith @"\Device\","Windows Local"
+      ,"")
+    , FilePath = iif(ObjectName startswith @"\Device\",""
+      ,ObjectName) 
+    , ActorUsername = Account
+    , Process = ProcessName
+    , ActingProcessId = toint(ProcessId)
+    , EventOriginalType = tostring(EventID)
+  | project-away ProcessName, EventID, ProcessId
+  | project-rename ActorUserId = SubjectUserSid
```
Comment (oshezaf) [inline Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml:63]: Add ActorUserSid as alias to ActorUserId

File: Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    , ActorUserType = iif(AccountType == "Machine", "Machine", "Regular")
+    , EventStartTime = TimeGenerated
+    , EventEndTime = TimeGenerated
+    , TargetFilePath = iif(ObjectName startswith @"\Device\",""
+      ,ObjectName) 
+    , TargetFilePathFormat = iif(ObjectName !startswith @"\Device\","Windows Local"
+      ,"")
+    , FilePath = iif(ObjectName startswith @"\Device\",""
+      ,ObjectName) 
+    , ActorUsername = Account
```
Comment (oshezaf) [inline Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml:58]: Add ActorWindowsUsername as alias to ActorUsername

File: Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    , Process = ProcessName
+    , ActingProcessId = toint(ProcessId)
+    , EventOriginalType = tostring(EventID)
+  | project-away ProcessName, EventID, ProcessId
+  | project-rename ActorUserId = SubjectUserSid
+    , EventOriginalSubType = Activity
+    , Dvc = Computer
```
Comment (oshezaf) [inline Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml:65]: DvcHostname = Computer. Dvc is an alias to DvcHostname

**Review by oshezaf — CHANGES_REQUESTED** [review]
- Don't forget to add sample data
- Additional fields:
  - SubjectLogonId ->ActorSessionId (based on the process schema which is more advanced)
  - Handle ID is important. Not sure what's the SecurityEvent field for this original field. Let's map to FileSessionId
  - Add DvcId = SourceComputerId,

File: Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    Default: false
+ParserQuery: |
+  let Parser=(disabled:bool=false)
+  {
+  SecurityEvent
+  | where not(disabled)
+  | project TimeGenerated, EventID, ObjectType, AccessMask, ProcessName, SubjectUserSid, AccountType, Account, Activity, Computer, ObjectName, ProcessId
+  | where EventID == 4663 
+    and ObjectType == "File"
+  | extend EventType = case(
```
[the remark this answers is not in the document]
Comment (KustoKing) [reply Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml:31]: OK

File: Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml (hunk 5 above)
Comment (KustoKing) [reply Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml:65]: OK

File: Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml (hunk 3 above)
Comment (KustoKing) [reply Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml:63]: OK

File: Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    , EventEndTime = TimeGenerated
+    , TargetFilePath = iif(ObjectName startswith @"\Device\",""
+      ,ObjectName) 
+    , TargetFilePathFormat = iif(ObjectName !startswith @"\Device\","Windows Local"
+      ,"")
+    , FilePath = iif(ObjectName startswith @"\Device\",""
+      ,ObjectName) 
+    , ActorUsername = Account
+    , Process = ProcessName
+    , ActingProcessId = toint(ProcessId)
```
[the remark this answers is not in the document]
Comment (KustoKing) [reply Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml:60]: OK

File: Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    , EventOriginalType = tostring(EventID)
+  | project-away ProcessName, EventID, ProcessId
+  | project-rename ActorUserId = SubjectUserSid
+    , EventOriginalSubType = Activity
```
[the remark this answers is not in the document]
Comment (KustoKing) [reply Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml:64]: 4663 - An attempt was made to access an object.

EventType is more valuable, so I remove this one.

File: Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+    , AccessMask == "0x20000", "MetadataAccessed"
+    , AccessMask == "0x4", "ObjectModified"
+    , AccessMask == "0x40", "ObjectDeleted"
+    , AccessMask == "0x40000", "MetadataModified"
+    , AccessMask == "0x6", "ObjectModified"
+    , AccessMask == "0x8", "MetadataAccessed"
+    , AccessMask == "0x80", "MetadataAccessed"
+    , AccessMask == "0x80000", "MetadataModified"
+    , "")
+    , ActingProcessName = ProcessName
```
[the remark this answers is not in the document]
Comment (KustoKing) [reply Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml:46]: OK

File: Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml (hunk 4 above)
Comment (KustoKing) [reply Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml:58]: OK

File: Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml (hunk 1 above)
Comment (KustoKing) [reply Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml:56]: OK

File: Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+    , ActorUserIdType = iif(SubjectUserSid startswith_cs "S-1-5-","SID","")
+    , ActorUsernameType = iif(Account contains "\\", "Windows", "")
+    , ActorUserType = iif(AccountType == "Machine", "Machine", "Regular")
+    , EventStartTime = TimeGenerated
+    , EventEndTime = TimeGenerated
+    , TargetFilePath = iif(ObjectName startswith @"\Device\",""
```
[the remark this answers is not in the document]
Comment (KustoKing) [reply Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml:52]: OK

File: Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml (hunk 2 above)
Comment (KustoKing) [reply Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml:47]: OK

File: Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let Parser=(disabled:bool=false)
+  {
+  SecurityEvent
+  | where not(disabled)
+  | project TimeGenerated, EventID, ObjectType, AccessMask, ProcessName, SubjectUserSid, AccountType, Account, Activity, Computer, ObjectName, ProcessId
+  | where EventID == 4663
```
[the remark this answers is not in the document]
Comment (KustoKing) [reply Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml:29]: In 4663 both rename and delete get logged as deleted in 4660 only deletes. When we want to split 4663 AccessMask 0x10000 in delete/rename we can use 4660 with the HandleId. But for now 4663 with AccessMask 0x10000 is delete.

File: Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (KustoKing) [reply Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml:58]: OK

File: Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+ParserName: ASimFileEventMicrosoftAuditEvent
+EquivalentBuiltInParser: _ASim_FileEvent_MicrosoftAuditEvent
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let Parser=(disabled:bool=false)
+  {
+  SecurityEvent
```
[the remark this answers is not in the document]
Comment (KustoKing) [reply Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml:26]: OK

**Comment by KustoKing** [comment]
> * Don't forget to add sample data
> * Additional fields:
>   
>   * SubjectLogonId ->ActorSessionId (based on the process schema which is more advanced)
>   * Handle ID is important. Not sure what's the SecurityEvent field for this original field. Let's map to FileSessionId
>   * Add DvcId = SourceComputerId,

All ready except DvcId because it is not always unique

File: Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml (hunk 3 above)
Comment (oshezaf) [reply Parsers/ASimFileEvent/ProductParsers/ASimFileEventMicrosoftAuditEvent.yaml:89]: You marked it as resolved by I can't see the update. Maybe something with how I see the PR and related files. There  should be a line 
extend ActorUserSid  = ActorUserId somewhere.

**Review by oshezaf — CHANGES_REQUESTED** [review]

**Pull request closed without merging**